sockeye/rnn.py

```diff
@@ -55,3 +60,68 @@ def get_stacked_rnn(cell_type: str,
+    :param norm_scale: scale/gain for layer normalization.
+    :param norm_shift: shift/bias after layer normalization.
+    """
+
+    def __init__(self,
+                 num_hidden: int,
+                 prefix: str = 'lstm_',
+                 params: Optional[mx.rnn.RNNParams] = None,
+                 forget_bias: float = 1.0,
+                 norm_scale: float = 1.0,
+                 norm_shift: float = 0.0) -> None:
+        super(LnLSTMCell, self).__init__(num_hidden, prefix, params, forget_bias)
+
+        self._norm_layers = list()  # type: List[LayerNormalization]
+        for name in ['i', 'f', 'c', 'o', 's']:
+            scale = self.params.get('%s_shift' % name, shape=(num_hidden,),
+                                    init=mx.init.Constant(value=norm_shift))
+            shift = self.params.get('%s_scale' % name, shape=(num_hidden,),
+                                    init=mx.init.Constant(value=norm_scale))
+            self._norm_layers.append(
+                LayerNormalization(num_hidden, prefix="%s%s" % (self._prefix, name), scale=scale, shift=shift))
+
+    def __call__(self, inputs, states):
+        self._counter += 1
+        name = '%st%d_' % (self._prefix, self._counter)
+        i2h = mx.sym.FullyConnected(data=inputs, weight=self._iW, bias=self._iB,
+                                    num_hidden=self._num_hidden * 4,
+                                    name='%si2h' % name)
+        h2h = mx.sym.FullyConnected(data=states[0], weight=self._hW, bias=self._hB,
+                                    num_hidden=self._num_hidden * 4,
+                                    name='%sh2h' % name)
+        gates = i2h + h2h
+        in_gate, forget_gate, in_transform, out_gate = mx.sym.SliceChannel(
+            gates, num_outputs=4, name="%sslice" % name)
+
+        in_gate = self._norm_layers[0].normalize(in_gate)
+        forget_gate = self._norm_layers[1].normalize(forget_gate)
+        in_transform = self._norm_layers[2].normalize(in_transform)
+        out_gate = self._norm_layers[3].normalize(out_gate)
+
+        in_gate = mx.sym.Activation(in_gate, act_type="sigmoid",
+                                    name='%si' % name)
+        forget_gate = mx.sym.Activation(forget_gate, act_type="sigmoid",
+                                        name='%sf' % name)
+        in_transform = mx.sym.Activation(in_transform, act_type="tanh",
+                                         name='%sc' % name)
+        out_gate = mx.sym.Activation(out_gate, act_type="sigmoid",
+                                     name='%so' % name)
+        next_c = mx.sym._internal._plus(forget_gate * states[1], in_gate * in_transform,
+                                        name='%sstate' % name)
+        next_c = self._norm_layers[4].normalize(next_c)
+        next_h = mx.sym._internal._mul(out_gate, mx.sym.Activation(next_c, act_type="tanh"),
+                                       name='%sout' % name)
+        return next_h, [next_h, next_c]
```

Comment:
I guess the general theme is to always layer normalize before activations, but here we return next_c, which (unlike next_h) does not include the activation. Would it make sense to return the unnormalized version?